### DISCUSSION TITLE
Agilent PNA-L N5230A not able to connect

### DISCUSSION THREAD

**Discussion by rizquierdol** [microsoft/Qcodes#4647] [Q&A]
Hi everyone

I have tried the QCoDeS Example with Keysight Network Analyzers, my equipment is an Agilent N5230A vector analyzer, however I am having some trouble with the code. I can confirm wether I am connected to the equipment by sending a IDN? query through Keysight IO suite. Below is the code and the error I receive. I have tried with the N5230C and N52xx modules but I got the same error.

![imagen](https://user-images.githubusercontent.com/114161197/191761685-6003d059-2a2d-4791-b8c6-39d727b3bce9.png)

The error:
![imagen](https://user-images.githubusercontent.com/114161197/191762981-8e7b1d7d-ceb3-472b-a11e-b0018f8aec72.png)
![imagen](https://user-images.githubusercontent.com/114161197/191763153-110bbc3e-2ab5-409e-a621-288fa21d78e4.png)
![imagen](https://user-images.githubusercontent.com/114161197/191763292-a625d16a-af28-4e41-b056-adf604e0c6cd.png)
![imagen](https://user-images.githubusercontent.com/114161197/191763481-3f4e74bc-019d-4fcf-bb5c-0b138f6c7a9a.png)

Many thanks

**Comment by jenshnielsen** [comment]
It looks like what happens is that during the setup the driver tries to get the "trace_catalog" from the instrument using the command `"CALC:PAR:CAT:EXT?"` which for some reason gives a timeout. I am not familiar with these pnas but perhaps you could try to 
send that command from Keysight IO and see if that gives an error too? It may also be worthwhile looking at the manual to see if that command should only be used in certain circumstances.
